tests/config/test_config.py

```diff
@@ -92,26 +92,6 @@ def test_load_filters_work_with_several_rules(template_two_roles_dict, test_file
                 resource_ids={"RootRoleTwo"},
                 resource_types={"AWS::IAM::Role"},
             ),
-            Failure(
-                granularity=RuleGranularity.RESOURCE,
-                reason="RootRoleTwo should not allow wildcard in principals or account-wide principals (principal: 'arn:aws:iam::123456789:user/someuser@bla.com')",
-                risk_value=RuleRisk.MEDIUM,
-                rule="PartialWildcardPrincipalRule",
-                rule_mode=RuleMode.BLOCKING,
-                actions=None,
-                resource_ids={"RootRoleTwo"},
-                resource_types={"AWS::IAM::Role"},
-            ),
-            Failure(
-                granularity=RuleGranularity.RESOURCE,
-                reason="RootRoleTwo should not allow wildcard in principals or account-wide principals (principal: 'arn:aws:iam::123456789:user/someuser@bla.com')",
-                risk_value=RuleRisk.MEDIUM,
-                rule="PartialWildcardPrincipalRule",
-                rule_mode=RuleMode.BLOCKING,
-                actions=None,
-                resource_ids={"RootRoleTwo"},
-                resource_types={"AWS::IAM::Role"},
-            ),
```

Comment:
This test was wrong as a result of iterating all principals and not some of them in GenericWildcardPrincipalRule. So we were adding false-positive.